test/server/guarddog_impl_test.cc

```diff
@@ -189,6 +189,48 @@ TEST_F(GuardDogMissTest, MegaMissTest) {
   unpet_dog = nullptr;
 }
 
+TEST_F(GuardDogMissTest, MissCountTest) {
+  // This tests a flake discovered in the MissTest where real timeout or
+  // spurious condition_variable wakeup causes the counter to get incremented
+  // more than it should be.
+  ON_CALL(time_source_, currentTime())
+      .WillByDefault(testing::Invoke([&]() {
+        return std::chrono::steady_clock::time_point(std::chrono::milliseconds(mock_time_));
+      }));
+  GuardDogImpl gd(stats_store_, config_miss_, time_source_);
+  auto sometimes_pet_dog = gd.createWatchDog(0);
+  for (unsigned long i = 0; i < 3; i++) {
```

Comment:
The test needs to verify that after we touch the watchdog the behavior resets back to baseline.  So it does this by executing the test steps twice, with the last step being the touch that should reset the behavior.

The third iteration is unnecessary, I'll remove it.